scripts/eurostat/gdp/import_data.py

```diff
@@ -0,0 +1,150 @@
+from preprocess_data import preprocess_df
+
+class EurostatGDPImporter:
+    """Pulls GDP data from the Eurostat database. Specifically, it processes the
+    data in the nama_10r_3gdp folder (available from the Eurostat website, linked
+    in README).
+
+    Attributes:
+        df: DataFrame (DF) with the cleaned data.
+    """
+    DATA_LINK = "https://ec.europa.eu/eurostat/estat-navtree-portlet-prod/BulkDownloadListing?file=data/nama_10r_3gdp.tsv.gz"
+    UNIT_CODES = {
+        "MIO_EUR":	         "Million euro",
+        "EUR_HAB":	         "Euro per inhabitant",
+        # "EUR_HAB_EU":	     "Euro per inhabitant in percentage of the EU average",
+        # "EUR_HAB_EU27_2020": "Euro per inhabitant in percentage of the EU27 (from 2020) average",
+        "MIO_NAC":	         "Million units of national currency",
+        "MIO_PPS":	         "Million purchasing power standards (PPS)",
+        # "MIO_PPS_EU27_2020": "Million purchasing power standards (PPS, EU27 from 2020)",
+        "PPS_HAB":	         "Purchasing power standard (PPS) per inhabitant",
+        # "PPS_EU27_2020_HAB": "Purchasing power standard (PPS, EU27 from 2020), per inhabitant",
+        # "PPS_HAB_EU":	     "Purchasing power standard (PPS) per inhabitant in percentage of the \
+        #                       EU average",
+        # "PPS_HAB_EU27_2020": "Purchasing power standard (PPS, EU27 from 2020), per inhabitant in \
+        #                       percentage of the EU27 (from 2020) average"
+    }
+    DESIRED_COLUMNS = ['geo', 'time'] + list(UNIT_CODES.keys())
+
+    # TODO (fpernice): Add a human-readable description of the codes to statVarObs
+    REGION_CODES = dict(json.loads(open('region_codes.json').read()))
+
+    def __init__(self):
+        self.raw_df = None
+        self.preprocessed_df = None
+        self.clean_df = None
+        print(self.REGION_CODES['EU27_2020'])
+
+    def download_data(self):
+        """Downloads raw data from Eurostat website and stores it in instance
+        data frame.
+        """
+        self.raw_df = pd.read_table(self.DATA_LINK)
+
+    def preprocess_data(self):
+        """Preprocesses instance raw_df and puts it into long format."""
+        if self.raw_df is None:
+            raise ValueError("Uninitialized value of raw data frame. Please "
+                             "check you are calling download_data before "
+                             "preprocess_data.")
+        self.preprocessed_df = preprocess_df(self.raw_df)
+
+    def clean_data(self):
+        """Drops unnecessary columns that are not needed for data import."""
+        if self.raw_df is None:
+            raise ValueError("Uninitialized value of processed data frame. "
+                             "Please check you are calling preprocess_data "
+                             "before clean_data.")
+        self.clean_df = self.preprocessed_df[self.DESIRED_COLUMNS]
+        self.clean_df = self.clean_df.replace(to_replace=':', value='')
```

Comment:
Please empirically check there are no notes in the Eurostat data.  Jin Zhou's Eurostat files sometimes had entries like "3458 : c" and ": c", where "c" is a note abbreviation.

Reply:
These notes are currently filtered out in Jin's script, and put into a different column. That column is then excluded in the line above: `self.clean_df = self.preprocessed_df[self.DESIRED_COLUMNS]`.